osbuild/util/containers.py

```diff
@@ -127,6 +148,30 @@ def container_source(image):
             os.symlink(image_filepath, tmp_source)
 
         image_source = f"{container_format}:{tmp_source}"
+        yield image_source
+
+
+@contextmanager
+def container_source(image):
+    image_filepath = image["filepath"]
+    container_format = image["data"]["format"]
+    image_name = image["data"]["name"]
+
+    if container_format not in ("dir", "oci-archive", "containers-storage"):
+        raise RuntimeError(f"Unknown container format {container_format}")
+
+    dispatcher = {
```

Comment:
I think that a plain `if` checking `container_format` and 3 branches would do the job and would be probably easier to read, but this works as well. :)

Reply:
That is my fault, sorry for this, I went a bit overboard here

Reply:
Haha I liked it, but I've changed it to a simple if/elif/else :)